pkg/cmd/checkendpoints/controller/metrics.go

```diff
@@ -0,0 +1,70 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/checkendpoints/trace"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+var (
+	registerMetrics sync.Once
+
+	endpointCheckCounter   *metrics.CounterVec
+	tcpConnectLatencyGauge *metrics.GaugeVec
+	dnsResolveLatencyGauge *metrics.GaugeVec
+)
+
+func RegisterMetrics(prefix string) {
+	registerMetrics.Do(func() {
+		endpointCheckCounter = metrics.NewCounterVec(&metrics.CounterOpts{
+			Name: prefix + "endpoint_check_count",
+			Help: "Report status of endpoint checks for each pod over time.",
+		}, []string{"endpoint", "tcpConnect", "dnsResolve"})
+
+		tcpConnectLatencyGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
```

Comment:
what do we use for infinity (no connection?)

Reply:
just an empty value `""`